File: docs/guides/swift/Sources/pushSetup.swift
import Foundation
#if os(Linux) // For linux interop
    import FoundationNetworking
#endif

import Core
import Ingestion

func pushSetup() async throws {
    do {
        let path = URL(string: #file)!.deletingLastPathComponent()
            .appendingPathComponent("records.json")
        let data = try Data(contentsOf: URL(fileURLWithPath: path.absoluteString))
        let records = try JSONDecoder().decode([PushTaskRecords].self, from: data)

        // use the region matching your applicationID
        let client = try IngestionClient(appID: "ALGOLIA_APPLICATION_ID", apiKey: "ALGOLIA_API_KEY", region: .us)

        // setting `watch` to `true` will make the call synchronous
        let resp = try await client.pushTask(
            taskID: "YOUR_TASK_ID",
            pushTaskPayload: PushTaskPayload(action: IngestionAction.addObject, records: records),
            watch: true
        )

        dump(resp)
    } catch {
        print(error)
    }
}

File: docs/guides/java/src/test/java/com/algolia/pushSetup.java
import com.algolia.api.IngestionClient;
import com.algolia.model.ingestion.*;
import com.fasterxml.jackson.databind.*;
import java.io.File;
import java.util.List;
import java.util.Map;

public class pushSetup {

  public static void main(String[] args) throws Exception {
    JsonNode content = new ObjectMapper().readTree(new File("records.json"));
    List<PushTaskRecords> records = new ObjectMapper().readerForListOf(Map.class).readValue(content);

    // use the region matching your applicationID
    IngestionClient client = new IngestionClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY", "ALGOLIA_APPLICATION_REGION");

    // setting `watch` to `true` will make the call synchronous
    WatchResponse resp = client.pushTask("YOUR_TASK_ID", new PushTaskPayload().setAction(Action.ADD_OBJECT).setRecords(records), true);

    System.out.println(resp);

    client.close();
  }
}
